contracts/container/contract.go

```diff
@@ -320,8 +322,15 @@ func Put(container []byte, signature interop.Signature, publicKey interop.Public
 	PutNamed(container, signature, publicKey, token, "", "")
 }
 
+// PutNamedOverloaded is the same as [Put] (and exposed as put from the contract via
+// overload), but allows named container creation via NNS contract.
+func PutNamedOverloaded(container []byte, signature interop.Signature, publicKey interop.PublicKey, token []byte, name, zone string) {
+	PutNamed(container, signature, publicKey, token, name, zone)
+}
+
 // PutNamed is similar to put but also sets a TXT record in nns contract.
 // Note that zone must exist.
+// DEPRECATED: use [PutNamedOverloaded].
```

Comment:
iiuc from https://github.com/nspcc-dev/neofs-contract/pull/451#issuecomment-2538854815, `put` is for everything. So, i'd deprecate this favor of `Put`, not `PutNamedOverloaded`

it's also interesting do contract manifests support `deprecated` markers? @roman-khimov

Reply:
> So, i'd deprecate this favor of Put

well, maybe. depends on how to read the code (methods that are written here will never be called on client side, it will be just "put"). hm, since we need to write here something, let it be ": call [Put] with the same args instead"